src/averageAges.js

```diff
@@ -55,7 +62,42 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => onlyWithSon
+    ? person.mother && person.sex === 'm'
+    : person.mother
+  );
+
+  const childMother = children.map(
+    child => Object.assign(
+      child, { motherBorn: findMotherBorn(people, child.mother) }
+    )
+  );
+
+  const isMother = childMother.filter(person => person.motherBorn);
+
+  const difference = isMother.map(
+    child => child.born - child.motherBorn
+  );
+  const sumDiff = difference.reduce((sum, age) => sum + age, 0);
+
+  return sumDiff / difference.length;
+}
+
+function findMotherBorn(people, motherName) {
+  const mother = people.find(person => person.name === motherName);
+
+  if (mother) {
+    return mother.born;
+  } else {
+    return false;
+  }
```

Comment:
avoid `if ()` statement
```suggestion
  return mother ? .... : ....
```